composer/datasets/streaming/dataset.py

```diff
@@ -392,6 +327,22 @@ def __iter__(self) -> Iterator[Any]:
             Iterator[Any]: Each sample.
         """
         Thread(target=self.download, daemon=True).start()
-
-        for idx in self._iter_ids():
-            yield self[idx]
+        num_workers = dist.get_local_world_size()
+        rank = dist.get_local_rank()
+        sbs = int(self._shuffle_buffer_size)
+        while self._batch_count < self.index.total_samples:
+            if self._batch_count < self._restored_batch_count:
+                self._batch_count += 1
+                yield None
```

Comment:
What's going on here? If we yield `None` will this impact the dataloader?

Reply:
ah good question - by setting `_restored_batch_count` to a nonzero number, we're signaling to the dataloader that anything before this point is not needed. In `trainer.py` in the main training loop, this allows us to trivially spin through `None` values without needing to calculate offsets, while keeping the spinning behavior for unsupported datasets.